modules/xml/filterx-func-format-xml.c

```diff
@@ -107,23 +107,27 @@ append_list(FilterXObject *key, FilterXObject *list, gpointer user_data)
 }
 
 static void
-_append_attribute(const char *key_str, const char *value_str, GString *buffer)
+_append_attribute(const char *key_str, GString *value_buffer, GString *buffer)
 {
   if (buffer->str[buffer->len - 1] == '>')
     g_string_overwrite(buffer, buffer->len - 1, " ");
   else
     g_string_append_c(buffer, ' ');
 
-  g_string_append_printf(buffer, "%s='%s'>", &key_str[1], value_str);
+  gchar *escaped_value = g_markup_escape_text(value_buffer->str, value_buffer->len);
+  g_string_append_printf(buffer, "%s='%s'>", &key_str[1], escaped_value);
```

Comment:
Everywhere we format attribute values, we should also escape `'`.

We have, for example, `str_repr_encode()` for that.